sync/sync.go

```diff
@@ -362,6 +375,31 @@ Register:
 
 }
 
+func (t *Task) isRevisionCountAndChainSame(addr string, fromClient, toClient *replicaClient.ReplicaClient) (bool, error) {
+	rwReplica, err := fromClient.GetReplica()
+	if err != nil {
+		return false, err
+	}
+
+	curReplica, err := toClient.GetReplica()
+	if err != nil {
+		return false, err
+	}
+
+	if rwReplica.RevisionCounter == curReplica.RevisionCounter {
+		if !reflect.DeepEqual(rwReplica.Chain[1:], curReplica.Chain[1:]) {
+			logrus.Errorf("Replica %v's chain not equal to RW replica %v's chain", toClient.GetAddress(), fromClient.GetAddress())
+			return false, nil
+		}
+
+		if err := t.client.VerifyRebuildReplica(rest.EncodeID(addr)); err != nil {
```

Comment:
doing this is required as controller as well need to change the state of replica in its structures..
good one :+1: